dpdispatcher/submission.py

```diff
@@ -519,17 +519,18 @@ def handle_unexpected_job_state(self):
             raise RuntimeError("job_state for job {job} is unknown".format(job=self))
 
         if job_state == JobStatus.terminated:
-            dlog.info(f"job: {self.job_hash} {self.job_id} terminated; restarting job")
-            if self.fail_count > 3:
-                raise RuntimeError("job:job {job} failed 3 times".format(job=self))
             self.fail_count += 1
+            dlog.info(f"job: {self.job_hash} {self.job_id} terminated;"
+                "fail_cout is {self.fail_count}; resubmitting job")
+            if self.fail_count > 3:
+                raise RuntimeError(f"job:{self.job_hash}failed 3 times.job_detail:{self}")
```

Comment:
```suggestion
                raise RuntimeError(f"job:{self.job_hash} failed 4 times. job_detail:{self}")
```
Also, I suggest to give more guides to users. In DP-GEN's issues, many users asked what to do for this error.

It's actually failing 4 times... Is it expected?

Reply:
Maybe we could print detailed debug information .  indicate the user how to debug like the  scirpt location, file directory etc ?

Reply:
Make sense. You can see many users don't know what to do:
deepmodeling/dpgen#434, deepmodeling/dpgen#398, deepmodeling/dpgen#394, deepmodeling/dpgen#360